find_meaningful_words.py

```diff
@@ -0,0 +1,127 @@
+    most_common_words = get_only_common_words(
+        words_frequencies_dict, number_of_most_common)
+    return most_common_words
+
+
+def turn_frequencies_into_words(frequencies_list):
+    return [x[0] for x in frequencies_list]
+
+
+def create_whole_frequencies_list(base_words, reference_words, reference_frequencies):
+    base_words_not_in_ref = set(base_words) ^ set(reference_words)
+    ref_freq_added_base_words = reference_frequencies + \
+        [(x, 1) for x in base_words_not_in_ref]
+    ref_words_whole = turn_frequencies_into_words(ref_freq_added_base_words)
+    return ref_words_whole, ref_freq_added_base_words
+
+
+def calculate_relative_from_frequencies(base_words, base_freq, ref_words_whole, ref_freq_added_base_words):
+
+    ref_ind = [ref_words_whole.index(x) for x in base_words]
+    base_relative = [(x[WORDS_PLACE_IN_FREQUENCY_LIST], x[FREQUENCY_PLACE_IN_FREQUENCY_LIST] /
+                      ref_freq_added_base_words[ref_ind[ind]][FREQUENCY_PLACE_IN_FREQUENCY_LIST]) 
+                      for ind, x in enumerate(base_freq)]
+    return base_relative
+
+
+def get_relative_frequencies(base_freq, ref_freq):
+    base_words = turn_frequencies_into_words(base_freq)
+    ref_words = turn_frequencies_into_words(ref_freq)
+    ref_words_whole, ref_freq_added_base_words = create_whole_frequencies_list(
+        base_words, ref_words, ref_freq)
+    base_relative = calculate_relative_from_frequencies(
+        base_words, base_freq, ref_words_whole, ref_freq_added_base_words)
+
+    return base_relative
+    pass
+
+
+def find_meaningful_words_in_file_from_other(base_path, referance_path, number_of_words):
+    base_words_frequencies = count_words_in_list(
+        get_words_only_in_string(read_file(base_path)))
+    referance_words_frequencies = count_words_in_list(
+        get_words_only_in_string(read_file(referance_path)))
+    base_relative_frequencies = get_relative_frequencies(
+        base_words_frequencies, referance_words_frequencies)
+    base_relative_frequencies = sort_frequencies_list(
+        base_relative_frequencies)
+    most_meaningful_words = get_only_common_words(
+        base_relative_frequencies, number_of_words)
+    return most_meaningful_words
+
+
+def main():
+    hamlet_path = "hamlet.txt"
+    # Find how many words are there in hamlet txt file.
+    hamlet_words_number = get_words_in_file(hamlet_path)
+    print(f"There are {hamlet_words_number} words in hamlet plat.")
+
+    # Find the 30 most common words.
+    how_many_common_words = 30
+    most_common_words = get_most_common_words_in_play(
+        hamlet_path, how_many_common_words)
+    print(
+        f"The {how_many_common_words} most common words in hamlet are - {most_common_words}")
+
+    # Having macbeth txt, could you find the most meaningful words in the hamlet file.
+    macbeth_path = "macbeth.txt"
+    hamlet_meaningful_words = find_meaningful_words_in_file_from_other(
```

Comment:
Yes, it's hamlets meaningful words. Maybe the documentation isn't clear enough?
Or because the variables aren't there?
I've added them now for less confusion

Reply:
I was being sarcastinc. at this point you switched from hamlet to mcbeth but the variables and outputs are still hamlet's

Reply:
Ah no, I'm not switching to Macbeth. I'm using Macbeth to find relevant words in Hamlet. That's why the variables names are still Hamlet's